docs/tutorials/using-typeorm.md

```diff
@@ -105,41 +108,66 @@ If preferred, it is possible to define a `type` to clean up the transaction meth
 ```
 
-### Unit Testing
-Use of TypeORM in the testing runtime is quite similar to using TypeORM in development, with the addition of appropriate placement of calls to set up and tear down the database schema.
+### Configuring TypeORM
 
-In `jest`, to set up the database once at the beginning of the test and tear down at the end:
-```javascript
-beforeAll(async () => {
-  testRuntime = await createTestingRuntime([DBOSAppClasses], "dbos-config.yaml");
-  await testRuntime.dropUserSchema(); // Optional
-  await testRuntime.createUserSchema();
-});
+:::info
+If you are using the [TypeORM template](#getting-started), this is done for you.
+:::
 
-afterAll(async () => {
-  await testRuntime.dropUserSchema();
-  await testRuntime.destroy();
-});
-```
+To enable TypeORM, you must set the `app_db_client` field in the [DBOS configuration file](../api-reference/configuration.md) to `typeorm`.
+You should also configure TypeORM migration commands.
+Here is an example of a configuration file set up for TypeORM:
 
-This will create the testing runtime, load TypeORM, and register the entities according to the decorators in `DBOSAppClasses`. The database will be configured according to the `database` section of the specified configuration `dbos-config.yaml` file:
 ```yaml
 database:
-  hostname: ${POSTGRES_HOST}
-  port: ${POSTGRES_PORT}
-  username: ${POSTGRES_USERNAME}
-  password: ${POSTGRES_PASSWORD}
-  app_db_name: ${POSTGRES_DATABASE}
-  system_database: 'opsys'
+  hostname: 'localhost'
+  port: 5432
+  username: 'postgres'
+  app_db_name: 'hello_typeorm'
+  password: ${PGPASSWORD}
   connectionTimeoutMillis: 3000
-  app_db_client: 'typeorm'
+  app_db_client: typeorm
+  migrate:
+    - npx typeorm migration:run -d dist/datasource.js
+  rollback:
+    - npx typeorm migration:revert -d dist/datasource.js
+runtimeConfig:
+  entrypoints:
+    - dist/src/operations.js
 ```
 
-The testing runtime can be used to invoke methods directly, or exercise handlers:
-```javascript
-  testRuntime.invoke(KVController, readUUID).readTxn("oaootest"),
-  const response = await request(testRuntime.getHandlersCallback()).get('/');
+Many TypeORM commands, such as those for [schema migration](#schema-management), require a TypeORM datasource file.
+To avoid managing your configuration in two places, we recommend this file use your DBOS configuration file as a source.
+Here is an example of a datasource file that does this:
+
+```typescript
+import { parseConfigFile } from '@dbos-inc/dbos-sdk/dist/src/dbos-runtime/config';
+import { TlsOptions } from 'tls';
+import { DataSource } from "typeorm";
+
+const [dbosConfig, ] = parseConfigFile();
+
+const AppDataSource = new DataSource({
+    type: 'postgres',
+    host: dbosConfig.poolConfig.host,
+    port: dbosConfig.poolConfig.port,
+    username: dbosConfig.poolConfig.user,
+    password: dbosConfig.poolConfig.password as string,
+    database: dbosConfig.poolConfig.database,
+    ssl: dbosConfig.poolConfig.ssl as TlsOptions,
```

Comment:
This is very clumsy.

Reply:
ORMEntities decorator would be preferable to me, because the entity list is code, not configuration.  Certainly not both things, with them getting out of sync with each other.

Reply:
How does that get the info to the migrations though?  They explicitly want a path to a datasource file.

Reply:
Sometimes it seems like TypeORM is very close but not quite there... so I'd be willing to write a little bit of code if that's what it takes to help the developer succeed.

Reply:
What are you suggesting?